### DISCUSSION TITLE
Apply @JsonNaming on serialization only

### DISCUSSION THREAD

**Discussion by flangam** [FasterXML/jackson-databind#3971] [Q&A] [answered]
Is there a way to use `@JsonNaming` **for serialization only**?  I'm ok with the default naming for deserialization (lower camel case).  But for serialization, I need to convert the field names to upper camel case like the following:

```
@JsonNaming(PropertyNamingStrategies.UpperCamelCaseStrategy.class)
public class MyDto {
   ...
}
```

Is this possible for serialization only?

Currently, to achieve this I'm using the following on **all fields** I want to convert.  For example:

```
@JsonAlias("name")
@JsonProperty("Name")
private String name;

@JsonAlias("message")
@JsonProperty("Message")
private String message;
```

But I'm looking for an efficient configuration without having to apply these 2 annotations on each field on each classes that I want to use this naming strategy.

**Comment by cowtowncoder** [comment, marked as answer]
No, unfortunately there is no way to apply `@JsonNaming` only on serialization; it will be used for transforming names in both directions.

Possible work-around would be to use 2 different mappers, and maybe configuring mix-in annotation for serialization case (to associate `@JsonNaming` on class(es).
Or using custom `AnnotationIntrospector` to block application of `@JsonNaming` for deserialization-side.
